Python-packages/covidcast-py/covidcast/geography.py

```diff
@@ -0,0 +1,63 @@
+import re
+
+import pandas as pd
+import pkg_resources
+
+COUNTY_CENSUS = pd.read_csv(
+    pkg_resources.resource_filename(__name__, "geo_mappings/county_census.csv"), dtype=str)
+MSA_CENSUS = pd.read_csv(
+    pkg_resources.resource_filename(__name__, "geo_mappings/msa_census.csv"), dtype=str)
+STATE_CENSUS = pd.read_csv(
+    pkg_resources.resource_filename(__name__, "geo_mappings/state_census.csv"), dtype=str)
+
+# Filter undesired rows from CSVs.
+# They're not removed from the files to keep them identical to rda files.
+COUNTY_CENSUS = COUNTY_CENSUS.loc[COUNTY_CENSUS.COUNTY != "0"]
+STATE_CENSUS = STATE_CENSUS.loc[STATE_CENSUS.STATE != "0"]
+MSA_CENSUS = MSA_CENSUS.loc[MSA_CENSUS.LSAD == "Metropolitan Statistical Area"]
+
+
+def fips_to_name(code, ignore_case=False, fixed=False, ties_method="first"):
+    return _lookup(code, COUNTY_CENSUS.FIPS, COUNTY_CENSUS.CTYNAME, ignore_case, fixed, ties_method)
+
+
+def cbsa_to_name(code, ignore_case=False, fixed=False, ties_method="first"):
+    return _lookup(code, MSA_CENSUS.CBSA, MSA_CENSUS.NAME, ignore_case, fixed, ties_method)
+
+
+def abbr_to_name(name, ignore_case=False, fixed=False, ties_method="first"):
+    return _lookup(name, STATE_CENSUS.ABBR, STATE_CENSUS.NAME, ignore_case, fixed, ties_method)
+
+
+def name_to_abbr(abbr, ignore_case=False, fixed=False, ties_method="first"):
+    return _lookup(abbr, STATE_CENSUS.NAME, STATE_CENSUS.ABBR, ignore_case, fixed, ties_method)
+
+
+def name_to_cbsa(name, ignore_case=False, fixed=False, ties_method="first", state=None):
+    df = MSA_CENSUS.loc[MSA_CENSUS.STATE == state.upper()] if state else MSA_CENSUS
+    return _lookup(name, df.NAME, df.CBSA, ignore_case, fixed, ties_method)
+
+
+def name_to_fips(name, ignore_case=False, fixed=False, ties_method="first", state=None):
+    df = COUNTY_CENSUS.loc[COUNTY_CENSUS.STNAME == abbr_to_name(state)[state].upper()] if state \
+        else COUNTY_CENSUS
+    return _lookup(name, df.CTYNAME, df.FIPS, ignore_case, fixed, ties_method)
+
+
+def _lookup(key, keys, values, ignore_case=False, fixed=False, ties_method="first"):
```

Comment:
In what cases would we *not* want to ignore case? I expect most US county/MSA names are not particularly case sensitive...

Reply:
I meant to ask that as well. I've pulled the defaults from the R code, but also thought going True might be better. I thought one argument against might be the edge case where someone sets `fixed=TRUE` but `ignore_case=TRUE` by default might lead to confusion, but I think the right implementation there is to have fixed override ignore_case with an info message. If we do want to go True, we should do in both R and Python.